addons/base_vat/models/res_partner.py

```diff
@@ -306,6 +307,26 @@ def check_vat_al(self, vat):
             return True
         return False
 
+    @api.model
+    def check_vat_do(self, vat):
+        """
+        Validate Dominican Republic tax numbers:
+        - RNC (9 digits, modulus-11 via stdnum.do.rnc)
+        - Cédula (11 digits, Luhn check)
+        """
+        number = ''.join(ch for ch in vat if ch.isdigit())
+
+        try:
+            if len(number) == 9:
+                rnc.validate(number)
+            elif len(number) == 11:
+                luhn.validate(number)
+            else:
+                return False
+            return True
+        except (ValidationError, InvalidChecksum):
+            return False
```

Comment:
We should use the [DO cedula module](https://github.com/arthurdejong/python-stdnum/blob/a1fdd5d5f39ec3d92da8fb1ab976d9e7175a571a/stdnum/do/cedula.py), there's apparently a whitelist. Let's take `check_vat_de` as an example and verify if it's either an rnc or cedula. That way we don't hardcode assumptions in the Odoo code and keep that the domain of `stdnum`:

```python
def check_vat_do(self, vat):
    is_valid_vat = stdnum.util.get_cc_module("do", "vat").is_valid
    is_valid_cedula = stdnum.util.get_cc_module("do", "cedula").is_valid
    return is_valid_vat(vat) or is_valid_cedula(vat)
```